packages/react-data-grid-react-window-grid/src/components/DataGrid/useDataGrid.ts

```diff
@@ -33,8 +31,14 @@ export const useDataGrid_unstable = (
     containerWidthOffset -= scrollbarWidth || 0;
   }
 
-  return useBaseState(
+  const baseState = useBaseState(
     { ...props, 'aria-rowcount': props.items.length, containerWidthOffset },
     ref
   );
+
+  return {
+    ...baseState,
+    bodyRef: props.bodyRef ?? React.useRef<VariableSizeGrid>(null),
+    headerRef: props.headerRef ?? React.useRef<VariableSizeList>(null),
```

Comment:
Hooks can't be called conditionally

Reply:
Thanks for the tips, will update it with useMergedRefs